pkg/extensions/sync/references/references.go

```diff
@@ -78,14 +83,32 @@ func (refs References) syncAll(ctx context.Context, localRepo, upstreamRepo,
 
 	// for each reference type(cosign/oci/oras reference)
 	for _, ref := range refs.referenceList {
+		supported, ok := refs.features.Get(ref.Name(), upstreamRepo)
+		if !supported && ok {
+			continue
+		}
+
 		syncedRefsDigests, err = ref.SyncReferences(ctx, localRepo, upstreamRepo, subjectDigestStr)
 		if err != nil {
+			// in the case of oci referrers, it will return 404 only if the repo is not found or refferers API is not supported
+			// no need to continue to make requests to the same repo
+			if ref.Name() == constants.OCI && errors.Is(err, zerr.ErrSyncReferrerNotFound) {
```

Comment:
There's no need to check zerr.ErrSyncReferrerNotFound, right? Both this if and the next one seem to do the same thing.

Reply:
the next one is !negated. I moved the second block here to make things more clear.